torch/testing/_internal/common_device_type.py

```diff
@@ -1356,6 +1374,17 @@ def only_fn(self, *args, **kwargs):
 
     return only_fn
 
+def onlyCUDAAndXPU(fn):
+    @wraps(fn)
+    def only_fn(self, *args, **kwargs):
+        if self.device_type not in ('cuda', 'xpu'):
+            reason = f"onlyCUDAAndXPU: doesn't run on {self.device_type}"
+            raise unittest.SkipTest(reason)
+
+        return fn(self, *args, **kwargs)
```

Comment:
Regarding this logic, we can refine `onlyOn` by extending it support device list. The code could be as follows.
```python
class onlyOn:

    def __init__(self, device_type: Union[str, List[str]]):
        self.device_types = []
        if isinstance(device_type, str):
            self.device_types.append(device_type)
        else:
            assert isinstance(device_type, list)
            self.device_types = device_type

    def __call__(self, fn):

        @wraps(fn)
        def only_fn(slf, *args, **kwargs):
            if slf.device_type not in self.device_types:
                reason = f"Only runs on {self.device_types}"
                raise unittest.SkipTest(reason)

            return fn(slf, *args, **kwargs)

        return only_fn
```

With the refined onlyOn, the above code could be as follows.
```python
def onlyCUDAAndXPU(fn):
    onlyOn(["cuda", "xpu"])(fn)
```

Reply:
Thanks, will do the update.